src/app/recover-funds/[address]/RecoverFundsButton.tsx

```diff
@@ -0,0 +1,172 @@
+import { useMemo } from 'react';
+import {
+  Address,
+  getL2Network,
+  L1ToL2MessageGasEstimator,
+  L1TransactionReceipt,
+} from '@arbitrum/sdk';
+import { Inbox__factory } from '@arbitrum/sdk/dist/lib/abi/factories/Inbox__factory';
+import { getBaseFee } from '@arbitrum/sdk/dist/lib/utils/lib';
+import { useNetwork, useSigner } from 'wagmi';
+import { getProviderFromChainId, getTargetChainId } from '../../../lib';
+import { BigNumber } from 'ethers';
+
+function RecoverFundsButton({
+  address,
+  balanceToRecover,
+  chainID,
+}: {
+  address: string;
+  balanceToRecover: BigNumber;
+  chainID: number;
+}) {
+  const [message, setMessage] = useState('');
+  const [loading, setLoading] = useState(false);
+  const { chain } = useNetwork();
+  const { data: signer } = useSigner();
+
+  const redeemButton = useMemo(() => {
+    if (!signer) return null;
+
+    if (chain?.id !== 1 && chain?.id !== 5) {
+      return (
+        <div>Unknown L1 chain id. This chain is not supported by this tool</div>
+      );
+    }
+
+    if (chain?.id !== chainID) {
+      return (
+        <div>
+          To recover funds, connect to chain ${chain?.id} (${chain?.name})
+        </div>
+      );
+    }
+
+    return (
+      <div className="recover-funds-form">
+        {loading && <div className="loading-block">Sending transaction...</div>}
+        <button
+          id="recover-button"
+          onClick={async () => {
+            setMessage('');
+
+            // Start transaction
+            const signerAddress = new Address(await signer.getAddress());
+            const aliasedSignerAddress = signerAddress.applyAlias();
+
+            // We instantiate the Inbox factory object to make use of its methods
+            const targetChainID = getTargetChainId(chainID);
+
+            if (!targetChainID) {
+              return;
+            }
+
+            const baseL2Provider = getProviderFromChainId(targetChainID);
+            const l2Network = await getL2Network(baseL2Provider);
+            const inbox = Inbox__factory.connect(
+              l2Network.ethBridge.inbox,
+              baseL2Provider,
+            );
+
+            // We estimate gas usage
+            const l1ToL2MessageGasEstimator = new L1ToL2MessageGasEstimator(
+              baseL2Provider,
+            );
+
+            if (!signer.provider) {
+              return;
+            }
+
+            // The estimateAll method gives us the following values for sending an L1->L2 message
+            //      (1) maxSubmissionCost: The maximum cost to be paid for submitting the transaction
+            //      (2) gasLimit: The L2 gas limit
+            //      (3) maxFeePerGas: The price bid per gas on L2
+            //      (4) deposit: The total amount to deposit on L1 to cover L2 gas and L2 call value
+            const gasEstimation = await l1ToL2MessageGasEstimator.estimateAll(
+              {
+                from: aliasedSignerAddress.value,
+                to: address,
```

Comment:
We should be able to choose a destination address, because the smart contract wallet might have a different address on L2 than it has on L1.